src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/Int128Converter.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal sealed class Int128Converter : JsonPrimitiveConverter<Int128>
+    {
+        private const int MaxFormatLength = 40;
+        private const int MaxEscapedFormatLength = MaxFormatLength * JsonConstants.MaxExpansionFactorWhileEscaping;
+
+        public Int128Converter()
+        {
+            IsInternalConverterForNumberType = true;
+        }
+
+        public override Int128 Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType != JsonTokenType.Number)
+            {
+                ThrowHelper.ThrowInvalidOperationException_ExpectedNumber(reader.TokenType);
+            }
+
+            return ReadCore(ref reader);
+        }
+
+        public override void Write(Utf8JsonWriter writer, Int128 value, JsonSerializerOptions options)
+        {
+            WriteCore(writer, value);
+        }
+
+        private static Int128 ReadCore(ref Utf8JsonReader reader)
+        {
+            if (reader.ValueLength > MaxEscapedFormatLength)
+            {
+                ThrowHelper.ThrowFormatException(NumericType.Int128);
+            }
+
+#if NET8_0_OR_GREATER
+            Span<byte> buffer = stackalloc byte[MaxFormatLength];
+#else
+            Span<char> buffer = stackalloc char[MaxFormatLength];
+#endif
+            int written = reader.CopyString(buffer);
```

Comment:
Given that you're reading a number token that can't contained escaping, I think it should be safe to just pass `reader.ValueSpan` to the `Int128` parsing method.

Reply:
Ah, I see now that you're using the same method for the property name parsing logic. Should be fine I suppose.